lib/components/pagination.vue

```diff
@@ -81,219 +87,286 @@
             >{{numberOfPages}}</a>
         </li>
 
+        <!-- Next page -->
         <li v-if="isActive(numberOfPages) || disabled" class="page-item disabled" aria-hidden="true">
-            <span class="page-link">&raquo;</span>
+            <span class="page-link" v-html="nextText"></span>
         </li>
         <li v-else class="page-item">
-            <a role="button"
+            <a :role="buttonRole"
                class="page-link"
                :aria-label="labelNextPage"
                tabindex="-1"
                @click.prevent="setPage($event, currentPage + 1)"
                @keydown.enter.prevent="setPage($event, currentPage + 1)"
                @keydown.space.prevent="setPage($event, currentPage + 1)"
-            ><span aria-hidden="true">&raquo;</span></a>
+            ><span aria-hidden="true" v-html="nextText"></span></a>
         </li>
 
     </ul>
 </template>
 
 <script>
-    function isVisible(el) {
-        return el && (el.offsetWidth > 0 || el.offsetHeight > 0);
-    }
+// Determine if an HTML element is visible - Faster than CSS check
+function isVisible(el) {
+    return el && (el.offsetWidth > 0 || el.offsetHeight > 0);
+}
 
-    export default {
-        data() {
-            return {
-                diff: 1,
-                showPrev: false,
-                showNext: false,
-                currentPage: this.value
-            };
+export default {
+    data() {
+        return {
+            offset: 1,
+            showFirstPage: false,
+            showFirstDots: false,
+            showLastPage: false,
+            showLastDots: false,
+            currentPage: this.value
+        };
+    },
+    computed: {
+        numberOfPages() {
+            const result = Math.ceil(this.totalRows / this.perPage);
+            return (result < 1) ? 1 : result;
         },
-        computed: {
-            numberOfPages() {
-                const result = Math.ceil(this.totalRows / this.perPage);
-                return (result < 1) ? 1 : result;
-            },
-            btnSize() {
-                return this.size ? `pagination-${this.size}` : '';
-            },
-            pageLinks() {
-                if (this.currentPage > this.numberOfPages) {
-                    this.currentPage = 1;
-                }
+        btnSize() {
+            return this.size ? `pagination-${this.size}` : '';
+        },
+        pageLinks() {
```

Comment:
Spent a while grocking that this computed prop is used as a range. Maybe we can add a comment, or just rename that a little more intuitively? `pageLinkRange` perhaps